syn/python/get_kge.py

```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/python3
+
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+# Parse a yosys area report and give a kGE equivalient
+
+import argparse
+
+
+def read_lib(lib_file_path):
+    with open(lib_file_path, 'r') as f:
+        lib_file = f.readlines()
+
+    cell_dict = {}
+    weighted_dict = {}
+    cell_name = ''
+    for line_idx, line in enumerate(lib_file):
```

Comment:
As a note, you could also iterate over the file here instead of slurping it at the start. The code would be something like
```python3
with open(lib_file_path, 'r') as lib_file:
    for line_idx, line in enumerate(lib_file):
        ...
```
Sadly, you gain another level of indentation, so I'm not convinced it's better. Also, it's not like we're streaming through GB of data, so it doesn't really matter to read everything first.

Reply:
Left as-is for now